src/DataSource/DoctrineCollectionDataSource.php

```diff
@@ -173,7 +173,9 @@ public function applyFilterText(Filter\FilterText $filter)
 		$exprs = [];
 
 		foreach ($filter->getCondition() as $column => $value) {
-			if ($filter->hasSplitWordsSearch() === FALSE) {
+			if($filter->isExactSearch()){
+				Criteria::expr()->eq($column, $value);
```

Comment:
You can't do this. You need to save the result into `$exprs[]` and call `continue` after it, just like you did in `DoctrineDataSource.php`